generators/generator-base.js

```diff
@@ -2141,8 +2111,170 @@ module.exports = class extends PrivateBase {
      */
     setupServerOptions(generator, context = generator, dest = context) {
         this.setupSharedOptions(generator, context, dest);
-        dest.enableTranslation = context.configOptions.enableTranslation || context.config.get('enableTranslation');
-        dest.testFrameworks = context.configOptions.testFrameworks;
+    }
+
+    loadOptions(options = this.options) {
+        // Parse options only once.
+        if (this.configOptions.optionsParsed) return;
+        this.configOptions.optionsParsed = true;
+
+        // Load runtime only options
+        this.configOptions.withEntities = options.withEntities;
+        this.configOptions.skipChecks = options.skipChecks;
+        this.configOptions.isDebugEnabled = options.debug;
+        this.configOptions.experimental = options.experimental;
+
+        // Load stored options
+        if (options.skipClient) {
+            this.skipClient = this.jhipsterConfig.skipClient = true;
+        }
+        if (options.skipServer) {
+            this.skipServer = this.jhipsterConfig.skipServer = true;
+        }
+        if (options.skipFakeData) {
+            this.jhipsterConfig.skipFakeData = true;
+        }
+        if (options.skipUserManagement) {
+            this.jhipsterConfig.skipUserManagement = true;
+        }
+        if (options.skipCheckLengthOfIdentifier) {
+            this.jhipsterConfig.skipCheckLengthOfIdentifier = true;
+        }
+        if (options.prettierJava) {
+            this.jhipsterConfig.prettierJava = true;
+        }
+        if (options.skipCommitHook) {
+            this.jhipsterConfig.skipCommitHook = true;
+        }
+
+        if (options.db) {
+            this.jhipsterConfig.databaseType = this.getDBTypeFromDBValue(this.options.db);
+            this.jhipsterConfig.devDatabaseType = options.db;
+            this.jhipsterConfig.prodDatabaseType = options.db;
+        }
+        if (options.auth) {
+            this.jhipsterConfig.authenticationType = options.auth;
+        }
+        if (options.uaaBaseName) {
+            this.jhipsterConfig.uaaBaseName = options.uaaBaseName;
+        }
+        if (options.searchEngine) {
+            this.jhipsterConfig.searchEngine = options.searchEngine;
+        }
+        if (options.build) {
+            this.jhipsterConfig.buildTool = options.build;
+        }
+        if (options.websocket) {
+            this.jhipsterConfig.websocket = options.websocket;
+        }
+        if (options.jhiPrefix) {
+            this.jhipsterConfig.jhiPrefix = options.jhiPrefix;
```

Comment:
@mshima : some tests on my side, it seems this.jhipsterConfig.jhiPrefix is always empty and is not initialized with `jhi` value.

My quick fix would be:
```
        if (options.jhiPrefix) {
            this.jhipsterConfig.jhiPrefix = options.jhiPrefix;
        } else {
            this.jhipsterConfig.jhiPrefix = 'jhi'
        }
```

But I know it's not the correct fix, because it should read the default value `jhi` from `generator-defaults.js`
 Can you help here plz ?

Reply:
@pascalgrimaud https://github.com/jhipster/generator-jhipster/pull/12063